src/Generating/Trees.cpp

```diff
@@ -367,40 +446,15 @@ void GetLargeAppleTreeImage(int a_BlockX, int a_BlockY, int a_BlockZ, cNoise & a
 {
 	int Height = 7 + a_Noise.IntNoise3DInt(a_BlockX, a_BlockY, a_BlockZ) % 4;
 
-	// Array with possible directions for a branch to go to.
-	const Vector3d AvailableDirections[] =
-	{
-		{ -1, 0, 0 }, { 0, 0, -1  },
-		{ -1, 0, 1 }, { -1, 0, -1 },
-		{ 1, 0, 1  }, { 1, 0, -1  },
-		{ 1, 0, 0  }, { 0, 0, 1   },
-
-		{ -0.5, 0, 0   }, { 0, 0, -0.5    },
-		{ -0.5, 0, 0.5 }, { -0.5, 0, -0.5 },
-		{ 0.5, 0, 0.5  }, { 0.5, 0, -0.5  },
-		{ 0.5, 0, 0    }, { 0, 0, 0.5     },
-
-		{ -1, 0.5, 0 }, { 0, 0.5, -1  },
-		{ -1, 0.5, 1 }, { -1, 0.5, -1 },
-		{ 1, 0.5, 1  }, { 1, 0.5, -1  },
-		{ 1, 0.5, 0  }, { 0, 0.5, 1   },
-
-		{ -0.5, 0.5, 0   },  { 0, 0.5, -0.5    },
-		{ -0.5, 0.5, 0.5 },  { -0.5, 0.5, -0.5 },
-		{ 0.5, 0.5, 0.5  },  { 0.5, 0.5, -0.5  },
-		{ 0.5, 0.5, 0    },  { 0, 0.5, 0.5     },
-
-	};
-
 	// Create branches
 	for (int i = 4; i < Height; i++)
 	{
 		// Get a direction for the trunk to go to.
-		Vector3d BranchStartDirection = AvailableDirections[static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockX, a_BlockY + i, a_BlockZ)) % ARRAYCOUNT(AvailableDirections)];
-		Vector3d BranchDirection = AvailableDirections[static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockX, a_BlockY / i, a_BlockZ)) % ARRAYCOUNT(AvailableDirections)] / 3;
+		Vector3d BranchStartDirection = availableBranchDirections()[static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockX, a_BlockY + i, a_BlockZ)) % availableBranchDirections().size()];
+		Vector3d BranchDirection = availableBranchDirections()[static_cast<size_t>(a_Noise.IntNoise3DInt(a_BlockX, a_BlockY / i, a_BlockZ)) % availableBranchDirections().size()] / 3;
```

Comment:
This brought me to an idea, why not just use a function that returns a single vector, based on the coords and the sequence?
`Vector3d pickBranchDirection(Vector3i a_BlockPos, int a_Seq)`
I think it'd make the code much more readable.

Reply:
Yeah, that does look better.